proxy.go

```diff
@@ -162,12 +162,87 @@ func (rp *reverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	requestDuration.With(s.labels).Observe(since)
 }
 
+func executeWithRetry(
+	ctx context.Context,
+	s *scope,
+	startTime time.Time,
+	retryNum int,
+	rp func(http.ResponseWriter, *http.Request),
+	rw ResponseWriterWithCode,
+	srw *statResponseWriter,
+	req *http.Request,
+	proxiedResponseDuration *prometheus.SummaryVec,
+) error {
+	numReplicas := len(s.host.replica.cluster.replicas)
```

Comment:
Can you leave the comment on top of this line why is there only retry when numReplicas > 1 ?

Reply:
Done